server/src/main/java/com/thoughtworks/go/server/security/GoExceptionTranslationFilter.java

```diff
@@ -80,6 +80,10 @@ private AuthenticationEntryPoint determineAuthenticationPoint(HttpServletRequest
         return point;
     }
 
+    private boolean isAnApiRequest(HttpServletRequest httpRequest) {
+        return httpRequest.getRequestURI().contains("/api/");
```

Comment:
I'd probably err on the side of doing a `startsWith` to avoid situations where you're accessing `files/:pipeline_name/:pipeline_counter/:stage_name/:stage_counter/:job_name/foo/api/index.html`